### PR TITLE
PLT-3193 Add channel notification preferences for push notifications

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
This PR is for the UI that makes use of the functionality to override the global notification preferences for each channel

#### Ticket Link
https://mattermost.atlassian.net/browse/PLT-3193

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Has UI changes
- [x] Includes text changes and localization file ([.../i18n/en.json](https://github.com/mattermost/platform/blob/master/i18n/en.json) and [.../webapp/i18n/en.json](https://github.com/mattermost/platform/tree/master/webapp/i18n/en.json)) updates

### CODE REVIEW

**Comment by jasonblais** [comment]
First two issues are fine to leave if fixing them impacts performance:

#### FUNCTIONALITY
1 - 
1) Set Account Settings > Email Notifications to Never
2) Go to Channel dropdown > Notification Preferences > Email notifications 
Observed: First option reads "Global default (Immediately)"
   - Changes to "Global default (Never)" refresh or after someone posts a message in the channel.
Expected: First option reads "Global default (Never)"

2 - It takes a few seconds for the channel push notification setting to take effect. This is fine if that's expected, just making sure it's not a bug.

#### UI

3 - Move "Mark Channel Unread" option to the bottom of the list

image

4 - Change `Mobile push notifications` title to `Send mobile push notifications`

5 - UI text for email notifications:

```
Title: Send email notifications
Options:
  - Global default ({On/Off})
  - On
  - Off
Help text: If on, email notifications are sent for mentions and direct messages when you are offline or away from Mattermost for more than 5 minutes. To change the global default, go to Account Settings > Notifications > Email notifications.
```

^functionality doesn't change, only the UI text does.

----

PS: There was a bug with username not showing for all push notifications -- reproduced on `master` so Eric will create a separate ticket for that.

**Comment by enahum** [comment]
@jasonblais as we discussed 3, 4, 5 are done

File: webapp/components/channel_notifications_modal.jsx [hunk 1]
```diff
@@ -375,8 +380,218 @@ export default class ChannelNotificationsModal extends React.Component {
...
+            user_id: currentUserId,
+            push: notifyLevel
+        };
+
+        updateChannelNotifyProps(data,
+            () => {
+                // YUCK
+                const member = ChannelStore.getMyMember(channelId);
+                member.notify_props.push = notifyLevel;
+                ChannelStore.storeMyChannelMember(member);
```
Comment (jwilander) [inline webapp/components/channel_notifications_modal.jsx]: Any way we can move this store interaction into the action?

Comment (enahum) [inline webapp/components/channel_notifications_modal.jsx]: Yeap, I think I can do that

**Comment by enahum** [comment]
@jwilander ready for re-review

**Comment by jwilander** [comment]
Ready to merge when build completes.

**Pull request merged**